src/main/java/org/jitsi/jicofo/rest/Health.java

```diff
@@ -271,4 +291,126 @@ private static int cacheStatus(int status)
+        {
+            // if there is no monitor task, this means we were stopped
+            if (this.monitorTask == null)
+            {
+                return;
+            }
+
+            // this monitoring was not stopped before the bad interval
+            // this means it takes too much time
+            try
+            {
+                ThreadMXBean threadMXBean
+                    = ManagementFactory.getThreadMXBean();
+                ThreadInfo[] threadInfos
+                    = threadMXBean.getThreadInfo(
+                        threadMXBean.getAllThreadIds(), 100);
+                StringBuilder dbg = new StringBuilder();
+                for (ThreadInfo threadInfo : threadInfos)
+                {
+                    dbg.append('"')
+                        .append(threadInfo.getThreadName()).append('"');
+
+                    Thread.State state = threadInfo.getThreadState();
+                    dbg.append("\n   java.lang.Thread.State: ")
+                        .append(state);
+
+                    if (threadInfo.getLockName() != null)
+                    {
+                        dbg.append(" on ").append(threadInfo.getLockName());
+                    }
+                    dbg.append('\n');
+
+                    StackTraceElement[] stackTraceElements
+                        = threadInfo.getStackTrace();
+                    for (int i = 0; i < stackTraceElements.length; i++)
+                    {
+                        StackTraceElement ste = stackTraceElements[i];
+                        dbg.append("\tat " + ste.toString());
+                        dbg.append('\n');
+                        if (i == 0 && threadInfo.getLockInfo() != null)
+                        {
+                            Thread.State ts = threadInfo.getThreadState();
+                            if (ts == Thread.State.BLOCKED
+                                || ts == Thread.State.WAITING
+                                || ts == Thread.State.TIMED_WAITING)
+                            {
+                                dbg.append("\t-  " + ts + " on "
+                                    + threadInfo.getLockInfo());
+                                dbg.append('\n');
+                            }
+                        }
+
+                        for (MonitorInfo mi
+                                : threadInfo.getLockedMonitors())
+                        {
+                            if (mi.getLockedStackDepth() == i) {
+                                dbg.append("\t-  locked " + mi);
+                                dbg.append('\n');
+                            }
+                        }
+                    }
+                    dbg.append("\n\n");
+                }
+                logger.error("Health check took "
+                    + (System.currentTimeMillis() - this.startedAt)
+                    + " ms. \n"
+                    + dbg.toString());
+            }
+            catch (Throwable t)
```

Comment:
What is the reason for catching Throwable here ?

Reply:
In case something went wrong obtaining threads info and printing. But yeah it doesn't make sense as it is in a thread we don't care about, we care about the result of the print.